lib/axon.ex

```diff
@@ -1810,12 +1810,32 @@ defmodule Axon do
       unrolling of RNN.
 
   """
-  @doc type: :recurrent
-  def lstm(%Axon{output_shape: shape} = x, units, opts \\ [])
+  def lstm(%Axon{} = x, units) when is_integer(units) and units > 0 do
```

Comment:
I think you need to add `@doc type: :recurrent` to all clauses. And then we usually do:

```elixir
@doc """
See `lstm/4`.
"""
@doc type: :recurrent
def lstm(%Axon{} = x, units) do
```

I typically tend to document the one with the biggest arity, but any is fine, as long as it is consitent in the API. You may need to add more examples to Axon.lstm docs too.